docs/source/cache_system.mdx

```diff
@@ -0,0 +1,72 @@
+<!--Copyright 2023 The HuggingFace Team. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# Optimum Neuron Compilation Cache System
+
+As you might have noticed while training your models on Trainium instances, Neuron cores work on models compiled to the `neff` format. 
+It means that training requires one or multiple compilation step(s) that can take a long time. 
+
+With [`torch-neuronx`](https://github.com/aws-neuron/aws-neuron-samples/tree/master/torch-neuronx), a Neuron-compatible  version of PyTorch, the compiled files are stored in a local directory once you performed compilation, usually `/var/tmp/neuron-compile-cache`. This way you do not need to perform compilation again when you start another training with similar parameters. 
+
+We go one step further with Optimum Neuron by plugging this local cache directory to the Hugging Face Hub. During training, the [`TrainiumTrainer`] will check if compilation files are available on the Hub and download them if they are found, allowing you to save both time and cost by skipping the compilation phase.
+
+## How the caching system works
+
+### Hash computation
+
+Many factors can trigger compilation among which: 
+
+- The model
+- The input shapes
+- The precision of the model, full-precision or bf16
+- The version of the compiler
+- The number of Neuron cores used
+
+Using all these parameters, we can compute a hash, and we can then compare hashes we have locally for our training session against hashes stored on the Hugging Face Hub, and act accordingly (download or push).
+
+
+### Using a custom cache repo on the Hugging Face Hub
+
+The official repo on the Hub storing compilation files is `aws-neuron/optimum-neuron-cache`. In this repo, we (the Optimum Neuron team) push compilation files for commonly used models so that it is publicly available and free to use for everyone. But there are two limitations:
```

Comment:
```suggestion
The repository for the public cache is `aws-neuron/optimum-neuron-cache`. This repository includes all precompiled files for commonly used models so that it is publicly available and free to use for everyone. But there are two limitations:
```